superset/assets/src/explore/controls.jsx

```diff
@@ -95,7 +95,7 @@ const D3_PERCENT_FORMAT_OPTIONS = [
   ['.3%', '.3% (12345.432 => 1234543.200%)'],
 ];
 
-const ROW_LIMIT_OPTIONS = [10, 50, 100, 250, 500, 1000, 5000, 10000, 50000];
+const ROW_LIMIT_OPTIONS = [-1, 10, 50, 100, 250, 500, 1000, 5000, 10000, 50000];
```

Comment:
As we are not adding handling of -1 in all widgets. we should not modify this array. Instead override options in  `visTypes.jsx`. the possible way to do is as follow around Line 1422
```
 controlOverrides: {
      y_axis_format: {
        label: t('Number format'),
      },
      row_limit: {
       default: -1,
       choices:  formatSelectOptions([-1, ...ROW_LIMIT_OPTIONS]
     }
    }
```
`...` is js syntax to expand array in-place